photon-ml/src/main/scala/com/linkedin/photon/ml/algorithm/RandomEffectCoordinate.scala

```diff
@@ -14,85 +14,131 @@
  */
 package com.linkedin.photon.ml.algorithm
 
+import scala.collection.Set
+
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.rdd.RDD
+
 import com.linkedin.photon.ml.constants.StorageLevel
 import com.linkedin.photon.ml.data.{KeyValueScore, LabeledPoint, RandomEffectDataSet}
-import com.linkedin.photon.ml.function.DiffFunction
+import com.linkedin.photon.ml.function.IndividualObjectiveFunction
 import com.linkedin.photon.ml.model.{DatumScoringModel, RandomEffectModel}
-import com.linkedin.photon.ml.normalization.NoNormalization
-import com.linkedin.photon.ml.optimization.game.{OptimizationTracker, RandomEffectOptimizationProblem,
-RandomEffectOptimizationTracker}
+import com.linkedin.photon.ml.optimization.game.{OptimizationTracker, RandomEffectOptimizationProblem, RandomEffectOptimizationTracker}
 import com.linkedin.photon.ml.supervised.model.GeneralizedLinearModel
-import org.apache.spark.broadcast.Broadcast
-import org.apache.spark.rdd.RDD
-
-import scala.collection.Set
 
 /**
  * The optimization problem coordinate for a random effect model
  *
+ * @tparam Function The type of objective function used to solve individual random effect optimization problems
```

Comment:
I went with `Objective`, it's more specific but still short (as per our offline conversation)